src/queryexecutor.js

```diff
@@ -0,0 +1,350 @@
+const {AddressResolver} = require('./addressresolver.js');
+
+class QueryExecutor {
```

Comment:
does this need to be a class?

Reply:
not really but no reason for it to not be